source/RoslynAnalyzers/Descriptors.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Octopus.RoslynAnalyzers
+{
+    public static class Descriptors
+    {
+        static DiagnosticDescriptor GetTestAnalyzerDescriptor(string id, string title, string? description = null)
+        {
+            return new DiagnosticDescriptor(id,
+                title,
+                title,
+                "Octopus.Testing",
+                DiagnosticSeverity.Warning,
+                false, // Testing analysers are disabled by default, these will only be enabled for test projects
+                description);
+        }
+
+        static readonly string noBaseClassesDescription = "Creating base classes for tests is a really good way to make tests neat, and 1 or 2 levels " +
+            "of inheritance is generally pretty easy to reason with. From past experience, these hierarchies quickly " +
+            "get out of control and it gets really hard to find where different things set up in the chain." +
+            Environment.NewLine +
+            "If you need to share logic and resources (that are expensive to set up) between multiple tests, xUnits " +
+            "IClassFixture and IAssemblyFixture might be your answer: https://xunit.net/docs/shared-context" +
+            Environment.NewLine +
+            "If you just need to share some logic, setup, or helpers between these tests please make use of builders " +
+            "or feel free to write a custom helper or context class." +
+            Environment.NewLine +
+            "And finally, if you want to run a bunch of the same tests with slightly different configuration, make use of " +
+            "xUnit's Theories with InlineData/ClassData/MemberData. These won't run in parallel though, so please use sparingly " +
+            "in integration test land." +
+            Environment.NewLine +
+            "Reach out to @team-core-blue if you need any help with strategies to test efficiently without base classes.";
+
+        public static DiagnosticDescriptor Oct2001NoIntegrationTestBaseClasses => GetTestAnalyzerDescriptor(
+            "OCT2001",
```

Comment:
Whys this starting from 2?

Reply:
Left 1 open for all of the analysers for the server code. Used 2 for testing so it's somewhat out of the way.